aws/resource_aws_batch_compute_environment.go

```diff
@@ -0,0 +1,362 @@
+	d.SetId(arn)
+
+	return resourceAwsBatchComputeEnvironmentRead(d, meta)
+}
+
+func resourceAwsBatchComputeEnvironmentRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).batchconn
+
+	ce, err := getComputeEnvironment(conn, d.Get("name").(string))
+	if err != nil {
+		return err
+	}
+	if ce == nil {
+		return fmt.Errorf("[WARN] Error reading Compute Environment: \"%s\"", err)
+	}
+	d.Set("arn", ce.ComputeEnvironmentArn)
+	d.Set("compute_resources", flattenComputeResource(ce.ComputeResources))
+	d.Set("name", ce.ComputeEnvironmentName)
+	d.Set("service_role_arn", ce.ServiceRole)
+	d.Set("state", ce.State)
+	d.Set("type", ce.Type)
+	return nil
+}
+
+func resourceAwsBatchComputeEnvironmentUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).batchconn
+
+	updateInput := &batch.UpdateComputeEnvironmentInput{
+		ComputeEnvironment: aws.String(d.Get("name").(string)),
+		ComputeResources:   updateComputeResource(d),
+		ServiceRole:        aws.String(d.Get("service_role_arn").(string)),
+		State:              aws.String(d.Get("state").(string)),
+	}
+	_, err := conn.UpdateComputeEnvironment(updateInput)
+	if err != nil {
+		return err
+	}
+	return resourceAwsBatchComputeEnvironmentRead(d, meta)
+}
+
+func resourceAwsBatchComputeEnvironmentDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).batchconn
+	sn := d.Get("name").(string)
+
+	// Check to make compute environment exists
+	ce, err := getComputeEnvironment(conn, sn)
+	if err != nil {
+		return err
+	}
+
+	if ce == nil {
+		log.Printf("[DEBUG] Compute Environment %q is already gone", sn)
+		return err
+	}
+
+	_, err = conn.UpdateComputeEnvironment(&batch.UpdateComputeEnvironmentInput{
+		ComputeEnvironment: aws.String(sn),
+		State:              aws.String(batch.CEStateDisabled),
+	})
+	if err != nil {
+		return err
+	}
+
+	// Wait until the Compute Environment is disabled
+	err = resource.Retry(5*time.Minute, func() *resource.RetryError {
+		log.Printf("[DEBUG] Trying to delete Compute Environment service %s", sn)
+		_, err = conn.DeleteComputeEnvironment(&batch.DeleteComputeEnvironmentInput{
+			ComputeEnvironment: aws.String(sn),
+		})
+		if err == nil {
+			return nil
+		}
+		return resource.RetryableError(err)
+	})
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{batch.CEStatusUpdating, batch.CEStatusDeleting},
+		Target:     []string{batch.CEStatusDeleted},
+		Refresh:    batchComputeEnvironmentRefreshFunc(conn, sn),
+		Timeout:    10 * time.Minute,
+		Delay:      10 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf(
+			"Error waiting for Compute Environment (%s) to be destroyed: %s",
+			sn, err)
+	}
+
+	d.SetId("")
+	return nil
+}
+
+func createComputeResource(d *schema.ResourceData) *batch.ComputeResource {
+	data := d.Get("compute_resources").([]interface{})[0]
```

Comment:
This seems to be the pattern used in other places in the repo, so I am sticking with it.